cli/awsbatch/utils.py

```diff
@@ -99,3 +99,36 @@ def is_job_array(job):
     :return: true if the job is an array, false otherwise
     """
     return "arrayProperties" in job and "size" in job["arrayProperties"]
+
+
+class S3Uploader(object):
+    """S3 uploader."""
+
+    def __init__(self, boto3_factory, s3_bucket, s3_folder=""):
+        """Constructor.
+
+        :param boto3_factory: initialized Boto3ClientFactory object
+        :param s3_bucket: S3 bucket to use
+        :param s3_bucket: S3 folder on which put the files (optional)
+        """
+        self.boto3_factory = boto3_factory
+        self.s3_bucket = s3_bucket
+        self.s3_folder = s3_folder
+
+    def put_file(self, file_path, key_name, timeout, s3_folder=None):
+        """
+        Upload a file to an s3 bucket.
+
+        :param file_path: file to upload
+        :param key_name: S3 key to create
+        :param timeout: S3 expiration time in seconds
+        :param s3_bucket: S3 folder on which put the files (optional, the default is the one passed to the constructor)
+        """
+        default_expiration = 30  # minutes
+        expires = datetime.datetime.now() + datetime.timedelta(minutes=default_expiration)
+        if timeout:
+            expires += datetime.timedelta(seconds=timeout)
```

Comment:
The expiration is going to be  `now + 30 min + timeout` that's not one would expect setting the timeout.